MediaBrowser.Model/Dto/MediaSourceInfo.cs

```diff
@@ -116,139 +117,91 @@ public MediaSourceInfo()
         public int? DefaultSubtitleStreamIndex { get; set; }
 
         [JsonIgnore]
-        public MediaStream VideoStream
+        public MediaStream? VideoStream
         {
             get
             {
-                foreach (var i in MediaStreams)
-                {
-                    if (i.Type == MediaStreamType.Video)
-                    {
-                        return i;
-                    }
-                }
-
-                return null;
+                return MediaStreams.Find(stream => stream.Type == MediaStreamType.Video);
             }
         }
 
         public void InferTotalBitrate(bool force = false)
         {
-            if (MediaStreams == null)
+            if (MediaStreams == null || (!force && Bitrate.HasValue))
             {
                 return;
             }
 
-            if (!force && Bitrate.HasValue)
-            {
-                return;
-            }
-
-            var bitrate = 0;
-            foreach (var stream in MediaStreams)
-            {
-                if (!stream.IsExternal)
-                {
-                    bitrate += stream.BitRate ?? 0;
-                }
-            }
+            int bitrate = MediaStreams
+                .Where(stream => !stream.IsExternal)
+                .Select(stream => stream.BitRate ?? 0)
+                .Sum();
```

Comment:
Performance is probably not a huge concern with this code, but we generally try to avoid using LINQ because it's allocating Enumerables, Method Groups etc. And our memory traffic is already quite painful.

Reply:
Ah, coming from a C++/Rust world, I had no idea about this!

Happy to leave this the way it was before if that's your preference?